source/core/src/main/com/csse3200/game/ui/terminal/commands/AddItemCommand.java

```diff
@@ -30,31 +33,24 @@ public boolean action(ArrayList<String> args) {
 			return false;
 		}
 		Entity player = ServiceLocator.getGameArea().getPlayer();
-		String itemName = args.get(0);
-		Entity item;
-		switch (itemName) {
-			case "hoe" -> item = ItemFactory.createHoe();
-			case "shovel" -> item = ItemFactory.createShovel();
-			case "can" -> item = ItemFactory.createWateringcan();
-			case "scythe" -> item = ItemFactory.createScythe();
-			case "sprinkler" -> item = ItemFactory.createSprinklerItem();
-			case "pump" -> item = ItemFactory.createPumpItem();
-			case "fence" -> item = ItemFactory.createFenceItem();
-			case "gate" -> item = ItemFactory.createGateItem();
-			case "chest" -> item = ItemFactory.createChestItem();
-			case "shipPart" -> item = ItemFactory.createShipPart();
-			default -> {
-				logger.debug("The provided item name does not exist");
-				return false;
-			}
+		StringJoiner str = new StringJoiner(" ");
+		for (String partOfName : args) {
+			str.add(partOfName);
+		}
+		String itemName = str.toString();
+		try {
+			Entity item = FactoryService.getItemFactories().get(itemName).get();
+			ServiceLocator.getGameArea().spawnEntity(item);
+			player.getComponent(InventoryComponent.class).addItem(item);
```

Comment:
Can this be moved outside the try? Since it's catching Exception, if any of those break it'll just catch it and report as incorrect item.

Reply:
Yup mb, I didn't have the return before but yes can be moved now